### PR TITLE
Lidar self collision filter, updated prefab

### PR DESCRIPTION
**Pull request by pijaro**
Signed-off-by: Piotr Jaroszek <piotr.jaroszek@robotec.ai>

### CODE REVIEW

File: Code/Source/Lidar/ROS2LidarSensorComponent.cpp [hunk 1]
```diff
@@ -102,14 +104,21 @@ namespace ROS2
         m_pointCloudPublisher.reset();
     }
 
-    void ROS2LidarSensorComponent::FrequencyTick()
-    {
+    void ROS2LidarSensorComponent::FrequencyTick() {
```
Comment (adamdbrw) [inline Code/Source/Lidar/ROS2LidarSensorComponent.cpp:107]: remove the urelated bracket change

File: Code/Source/Lidar/LidarRaycaster.h [hunk 2]
```diff
@@ -22,10 +23,11 @@ namespace ROS2
         //! in real sensors.
         //! @param directions Directions in which to shoot rays. These should be generated from Lidar configuration.
         //! @param distance Maximum distance for ray-casting.
+        //! @param parentId EntityId for self collision filtering.
         //! @return Hits of raycast. The returned vector size can be anything between zero and size of directions.
         //! No hits further than distance will be reported.
         // TODO - different starting points for rays, distance from reference point, noise models, rotating mirror sim, other
         // TODO - customized settings. Encapsulate in lidar definition and pass in constructor, update transform.
-        AZStd::vector<AZ::Vector3> PerformRaycast(const AZ::Vector3& start, const AZStd::vector<AZ::Vector3>& directions, float distance);
+        AZStd::vector<AZ::Vector3> PerformRaycast(const AZ::Vector3& start, const AZStd::vector<AZ::Vector3>& directions, float distance, const AZ::EntityId& parentId);
```
Comment (adamdbrw) [inline Code/Source/Lidar/LidarRaycaster.h:31]: Add a setter instead - this is done once per activate

File: Code/Source/Lidar/ROS2LidarSensorComponent.cpp [hunk 3]
```diff
@@ -102,14 +104,21 @@ namespace ROS2
...
-        const auto directions = LidarTemplateUtils::PopulateRayDirections(m_lidarModel, entityTransform->GetWorldTM().GetEulerRadians());
+        const auto directions = LidarTemplateUtils::PopulateRayDirections(m_lidarModel,
+                                                                          entityTransform->GetWorldTM().GetEulerRadians());
         AZ::Vector3 start = entityTransform->GetWorldTM().GetTranslation();
-        start.SetZ(start.GetZ() + 1.0f);
-        m_lastScanResults = m_lidarRaycaster.PerformRaycast(start, directions, distance);
+        start.SetZ(start.GetZ());
+
+        // If self collision entity is not set, use this one
+        if (!m_selfColliderEntityId.IsValid())
```
Comment (adamdbrw) [inline Code/Source/Lidar/ROS2LidarSensorComponent.cpp:116]: would be best to communicate to the user that this is optional - or set the self by default. I will design the feature.

Comment (pijaro) [reply Code/Source/Lidar/ROS2LidarSensorComponent.cpp]: I added a hint to the description

**Pull request merged**